site/app/controllers/admin/ConfigurationController.php

```diff
@@ -184,16 +184,15 @@ public function updateConfiguration(): Response {
             $entry = $entry === "true" ? true : false;
         }
 
-        if ($name === 'forum_enabled') {
-            if ($entry == 1) {
-                if ($this->core->getAccess()->canI("forum.modify_category")) {
-                    $categories = ["General Questions", "Homework Help", "Quizzes" , "Tests"];
-                    $rows = $this->core->getQueries()->getCategories();
-
-                    foreach ($categories as $category) {
-                        if (ForumUtils::isValidCategories($rows, -1, array($category))) {
-                            $this->core->getQueries()->addNewCategory($category);
-                        }
+        if ($name === 'forum_enabled' && $entry == 1) {
+            $categories = ["General Questions", "Homework Help", "Quizzes" , "Tests"];
+            $rows = $this->core->getQueries()->getCategories();
+
+            // Only create default categories when the forum is first enabled
+            if (empty($rows)) {
+                foreach ($categories as $category) {
+                    if (!ForumUtils::isValidCategories($rows, -1, array($category))) {
```

Comment:
We can get rid of this check, as the function will always trivially return false with `$rows` being empty.